Automattic/jetpack
# Add AMP compatibility to Slideshow shortcode

#### Changes proposed in this Pull Request:

* Adds AMP compatibility to the Slideshow shortcode
* Reuses the Slideshow block [render method](https://github.com/Automattic/jetpack/blob/34e9a2337a70ceec7a467a87bf04a83821a96397/extensions/blocks/slideshow/slideshow.php#L54)
* No change on non-AMP endpoints

#### Is this a new feature or does it add/remove features to an existing part of Jetpack?
* Enhances an existing feature

#### Testing instructions:
1. Activate the [AMP plugin](https://wordpress.org/plugins/amp/)
2. In AMP Settings, select Transitional mode
3. Create a new post 
4. Add a Shortcode block
5. Enter this, with ids from your environment:
```html
[gallery ids="1691,766,760" type="slideshow"]
```
The attribute `type="slideshow"` is what triggers the code below.

6. Preview the front-end of the AMP URL:

![amp-button](https://user-images.githubusercontent.com/4063887/78599669-28a98280-7817-11ea-8da1-9cc9d29ea79e.png)

7. Expected: there's an `<amp-carousel>`, and it displays captions if the images have them:
![slideshow](https://user-images.githubusercontent.com/4063887/78598819-9c4a9000-7815-11ea-8c42-6d9059cefe9a.gif)

This won't look exactly like the non-AMP endpoint, but it should have the same images.

8. Preview the non-AMP URL:

![amp-button](https://user-images.githubusercontent.com/4063887/78599696-3959f880-7817-11ea-9674-1fd76b6e53ba.png)

9. Checkout the `master` branch
10. Preview the non-AMP URL in another tab
11. Expected: the non-AMP URL looks the same with this PR and without

#### Proposed changelog entry for your changes:
* Add AMP compatibility to Slideshow shortcode
